File: trunk/ShadowFrameworkLite20/src/shadow/renderer/SFReferenceNode.java
package shadow.renderer;


import shadow.math.SFMatrix3f;
import shadow.math.SFVertex3f;

public class SFReferenceNode extends SFAbstractReferenceNode implements SFNode{
	
	@Override
	public SFNode copyNode() {
		SFReferenceNode reference=new SFReferenceNode();
		cloneOn(reference);
		return reference;
	}

	protected void cloneOn(SFReferenceNode reference) {
		for (SFNode sfNode : nodes) {
			reference.addNode(sfNode.copyNode());
		}
		SFVertex3f tmpV=new SFVertex3f();
		SFMatrix3f tmpM=new SFMatrix3f();
		this.transform.getPosition(tmpV);
		reference.transform.setPosition(tmpV);
		this.transform.getOrientation(tmpM);
		reference.transform.setOrientation(tmpM);
	}
}

File: ShadowFrameworkJS/src/shadow/renderer/SFReferenceNode.js


function SFReferenceNode(){
	this.nodes=new Array();
	this.initialize();
}

inherit(SFReferenceNode,SFAbstractReferenceNode);


SFReferenceNode.prototype["copyNode"]=function(){
	var reference=new SFReferenceNode();
	this.cloneOn(reference);
	return reference;
};


SFReferenceNode.prototype["cloneOn"]=function(reference){
	for(var i=0;i<this.nodes.length;i++){
		reference.addNode(this.nodes[i].copyNode());
	}
	var tmpV=new SFVertex3f();
	var tmpM=new SFMatrix3f();
	this.transform.getPosition(tmpV);
	reference.transform.setPosition(tmpV);
	this.transform.getOrientation(tmpM);
	reference.transform.setOrientation(tmpM);
};

